crates/telio-wg/src/link_detection/mod.rs

```diff
@@ -315,48 +256,26 @@ impl StateDecision {
 }
 
 struct State {
-    stats: BytesAndTimestamps,
+    stats: Arc<Mutex<BytesAndTimestamps>>,
     variant: StateVariant,
 }
 
 impl State {
     const POSSIBLE_DOWN_DELAY: Duration = Duration::from_secs(3);
 
-    fn new(rx_bytes: Option<u64>, tx_bytes: Option<u64>, node_state: NodeState) -> Self {
-        let stats = BytesAndTimestamps::new(rx_bytes, tx_bytes);
-        if matches!(node_state, NodeState::Connected) {
-            State {
-                stats,
-                variant: StateVariant::Up,
-            }
-        } else {
-            State {
-                stats,
-                variant: StateVariant::Down,
-            }
+    fn new(stats: Arc<Mutex<BytesAndTimestamps>>, rtt: Duration) -> Self {
```

Comment:
```suggestion
    fn new(stats: Arc<Mutex<BytesAndTimestamps>>, cfg_max_allowed_rtt: Duration) -> Self {
```

nit: as `rtt` is representing configured maximum allowed RTT value which is used to declare link down -> would be cool to update the name to make it slightly more clear.

Reply:
renamed